### ISSUE TITLE
does runai.ga support BatchNorm and Conv layers? 

### ISSUE THREAD

**Issue by iljimae0418** [run-ai/runai#19]
Right now, I'm using SeparableConv2D and Conv2D, with Dropout and BatchNormalization 

does runai.ga support gradient accumulation for these functionalities? 

I've seen some other wrapper functions that do not support BatchNorm, so I was wondering if this supports BatchNorm and what not. 

On what keras functionalities has runai.ga been tested on?

**Comment by levosos** [comment]
Hi, and thanks for reaching out!

`runai.ga` uses the gradients of the model and affects them only. 
Therefore, all layers are implicitly supported, and in particular the ones that you specified.
As you already mentioned, there are logical/algorithmic implications, though.

Batch normalization will happen on every mini-batch separately, and not on the global batch.

For example, BN will happen 4 times if you are using GA with 4 steps, and not once on the global batch (the total 4 mini-batches).

`runai.ga` has unit-tests that can be seen [here](https://github.com/run-ai/runai/blob/master/tests/test_ga_keras.py), as well as tested examples that can be seen [here](https://github.com/run-ai/runai/tree/master/examples/ga/keras).
